### PR TITLE
Implement Access-code API

### LINKED ISSUE #797: Implement access code API to /tickets/access-codes/create
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [x] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-frontend

### PR DESCRIPTION
**Pull request by srv-twry**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Implements access code API to /tickets/access-codes/create

#### Changes proposed in this pull request:
Implements the access code creation.
Add access codes relationship in user model

Fixes #797

### CODE REVIEW

**Comment by CosmicCoder96** [comment]
@srv-twry Follow up at https://github.com/fossasia/open-event-server/issues/4711

**Comment by CosmicCoder96** [comment]
@srv-twry any updates on this ?

**Comment by srv-twry** [comment]
Yes, i am working on it. The blockage from the server has been removed. 
I am updating the PR, but facing a small issue: The submit button isn't showing up for me even though it's in the hbs file. It also doesn't appear for me in the case of discount code.

discnt
temp

Any clue why this could be the case ?

**Comment by srv-twry** [comment]
@CosmicCoder96 
It appears when I remove the access-link element above the submit button.

**Comment by CosmicCoder96** [comment]
@srv-twry is it working now ?

File: app/components/forms/events/view/create-access-code.js [hunk 1]
```diff
@@ -141,8 +141,9 @@ export default Component.extend(FormMixin, {
     return location.origin + this.get('router').urlFor('public', params['events.view'].event_id, { queryParams: { access_code: this.get('data.code') } });
   }),
   actions: {
-    submit() {
+    submit(data) {
       this.onValid(() => {
+        this.sendAction('save', data);
```
Comment (CosmicCoder96) [inline app/components/forms/events/view/create-access-code.js]: use the `save` action directly instead of sending it.

**Comment by srv-twry** [comment]
@CosmicCoder96 There are still a few things to take care of. It will be ready for review by evening.

**Comment by srv-twry** [comment]
ezgif com-video-to-gif


@CosmicCoder96 @niranjan94 @ritikamotwani @pradeepgangwar Please review :)

File: app/components/forms/events/view/create-access-code.js [hunk 2]
```diff
@@ -100,49 +89,47 @@ export default Component.extend(FormMixin, {
...
+      } else {
+        this.set('allTicketTypesChecked', false);
+      }
+    },
+    submit(data) {
       this.onValid(() => {
+        data.save()
+          .then(() => {
+            this.get('notify').success(this.get('l10n').t('Access code has been successfully created.'));
+            this.get('router').transitionTo('events.view.tickets.access-codes');
```
Comment (srv-twry) [inline app/components/forms/events/view/create-access-code.js]: @CosmicCoder96 @niranjan94 Need your help here. When I redirect the user back to the access-codes list after creation of the access-codes, the list isn't updated. I need to reload the page in order for the list to be updated. 
How to resolve this ?

**Comment by srv-twry** [comment]
@pradeepgangwar @ritikamotwani @mayank8318 Please review. I am unable to fix the issue with the list not getting refreshed but this also occurs at other sections of the project. so may be we could fix it later.

**Comment by ritikamotwani** [comment]
@srv-twry the unit tests for controllers have currently been removed.

**Comment by srv-twry** [comment]
@ritikamotwani Removed them.

File: app/components/forms/events/view/create-access-code.js [hunk 3]
```diff
@@ -2,8 +2,10 @@ import Component from '@ember/component';
 import { computed } from '@ember/object';
 import $ from 'jquery';
 import FormMixin from 'open-event-frontend/mixins/form';
+import { inject as service } from '@ember/service';
 
 export default Component.extend(FormMixin, {
+  router: service(),
```
Comment (niranjan94) [inline app/components/forms/events/view/create-access-code.js]: Remove this declaration here. This injected globally.

File: package.json [hunk 4]
```diff
@@ -82,6 +82,7 @@
     "ember-metrics": "^0.12.1",
     "ember-models-table": "^2.5.0",
     "ember-moment": "^7.7.0",
+    "ember-multiselect-checkboxes": "^0.11.1",
```
Comment (niranjan94) [inline package.json]: Do we really need an addon for something as simple and straightforward as checkboxes ?

**Pull request merged**